From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Maciej Patelczyk <maciej.patelczyk@intel.com>
Date: Tue, 13 Jan 2026 13:19:56 +0100
Subject: [PATCH] fixup! drm/xe/eudebug: Prelim rework for Xe EU debug

VM_BIND extension EUDEBUG_ATTACH to use prelim macros.

Signed-off-by: Maciej Patelczyk <maciej.patelczyk@intel.com>
(cherry-picked from commit e1940353a92b00ea07a0a8316ee14e5060e25290 eudebug-v3-xe3p-prelim)
Signed-off-by: Bommu Krishnaiah <krishnaiah.bommu@intel.com>
---
 drivers/gpu/drm/xe/xe_vm.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xe/xe_vm.c b/drivers/gpu/drm/xe/xe_vm.c
index 03bd193de..a3c109121 100644
--- a/drivers/gpu/drm/xe/xe_vm.c
+++ b/drivers/gpu/drm/xe/xe_vm.c
@@ -3723,8 +3723,12 @@ typedef int (*xe_vm_bind_op_user_extension_fn)(struct xe_device *xe,
 					       struct drm_gpuva_ops *ops,
 					       u32 operation, u64 extension);
 
-static const xe_vm_bind_op_user_extension_fn vm_bind_op_extension_funcs[] = {
-	[PRELIM_XE_VM_BIND_OP_EXTENSIONS_ATTACH_DEBUG] = prelim_vm_bind_op_ext_attach_debug,
+static const __maybe_unused xe_vm_bind_op_user_extension_fn vm_bind_op_extension_funcs[] = {
+#define MAKE_TABLE_IDX(id)[PRELIM_XE_VM_BIND_OP_EXTENSIONS_##id & PRELIM_DRM_XE_VALUE_MASK]
+
+	MAKE_TABLE_IDX(ATTACH_DEBUG) = prelim_vm_bind_op_ext_attach_debug,
+
+#undef MAKE_TABLE_IDX
 };
 
 #define MAX_USER_EXTENSIONS	16
@@ -3746,12 +3750,13 @@ static int vm_bind_op_user_extensions(struct xe_device *xe,
 		return -EFAULT;
 
 	if (XE_IOCTL_DBG(xe, ext.pad) ||
-	    XE_IOCTL_DBG(xe, ext.name >=
+	    XE_IOCTL_DBG(xe, ext.name != PRELIM_XE_VM_BIND_OP_EXTENSIONS_ATTACH_DEBUG) ||
+	    XE_IOCTL_DBG(xe, PRELIM_DRM_XE_VALUE(ext.name) >=
 			 ARRAY_SIZE(vm_bind_op_extension_funcs)))
 		return -EINVAL;
 
-	err = vm_bind_op_extension_funcs[ext.name](xe, xef, ops,
-						   operation, extensions);
+	err = vm_bind_op_extension_funcs[PRELIM_DRM_XE_VALUE(ext.name)](xe, xef, ops,
+									operation, extensions);
 	if (XE_IOCTL_DBG(xe, err))
 		return err;
 
-- 
2.43.0

